[TNT-245] 트레이니 식단 기록 확인 화면 구현

SeonJeongk opened the pull request:
## 📝 작업 내용

- Closes #91 

- 식단 기록 확인 화면 UI 및 API 연동까지 구현했습니다.
- 현재 트레이니 홈 화면 API 연동이 안 된 상태라, 연동한 후에 `식단 기록일 때만 화면 이동`할 수 있도록 예외처리 해두겠습니다!


## 📸 실행 화면
| 사진 있을 때 | 사진 없을 때 |
|-------------|-------------|
| ![사진 있을 때](https://github.com/user-attachments/assets/8bcc14f3-d9d4-45b6-ad7a-5c060ec7962c) | ![사진 없을 때](https://github.com/user-attachments/assets/314d4c93-ff78-49e6-b135-7d04ecb9e8a8) |





## 🙆🏻 리뷰 요청 사항
없습니다!


## 👀 레퍼런스

SeonJeongk committed:
[TNT-245] feat: 홈화면과 식단 기록 확인 화면 네비게이션 연결

SeonJeongk committed:
[TNT-245] move: 패키지 분리

SeonJeongk committed:
[TNT-245] feat: 식단 기록 상세 화면 UI 및 네비게이션 id 넘겨주도록 수정

SeonJeongk committed:
[TNT-245] feat: 식단 기록 조회 API 구현

SeonJeongk committed:
[TNT-245] fix: dietId 데이터 타입 수정

SeonJeongk committed:
[TNT-245] feat: 식단 기록 조회 API 연동

SeonJeongk committed:
[TNT-245] fix: DateTimeFormatter yyyy-MM-dd'T'HH:mm:ss 형식으로 수정

SeonJeongk committed:
[TNT-245] fix: 기록 id 데이터 타입 변경

SeonJeongk committed:
[TNT-245] fix: 식단 기록 이미지 오류 수정

hoyahozz commented on feature/trainee/mealrecord/src/main/java/co/kr/tnt/trainee/mealrecord/detail/TraineeMealRecordDetailContract.kt:
```diff
@@ -0,0 +1,38 @@
+        val mealType: String = "",
+        val memo: String = "",
+    ) : UiState {
+        private val dateFormatter = DateFormatter()
```
`Date` 변환은 컴포저블에서 진행해도 괜찮을 것 같습니다 ~!

hoyahozz commented on feature/trainee/mealrecord/src/main/java/co/kr/tnt/trainee/mealrecord/detail/TraineeMealRecordDetailContract.kt:
```diff
@@ -0,0 +1,38 @@
+        val id: Long = 0,
+        val image: String? = null,
+        val date: LocalDateTime = LocalDateTime.now(),
+        val mealType: String = "",
```
여기서부터 `MealType` 클래스 사용하는건 어려웠나요?

SeonJeongk replied:
https://github.com/YAPP-Github/TnT-Android/pull/93/commits/67e530ffbcb66d5b750fcedf71f50d9b710f8650

리뷰 모두 수정했습니다!

hoyahozz reviewed (approved):
고생하셨습니다 ~! 👍👍👍

SeonJeongk committed:
[TNT-245] fix: mealType을 MealType 클래스로 수정

SeonJeongk committed:
[TNT-245] Merge branch 'develop' into feature/TNT-245

# Conflicts:
#	data/network/src/main/java/co/kr/data/network/service/ApiService.kt
#	data/network/src/main/java/co/kr/data/network/source/TraineeRemoteDataSource.kt
#	domain/src/main/java/co/kr/tnt/domain/repository/TraineeRepository.kt

SeonJeongk committed:
[TNT-245] Merge branch 'develop' into feature/TNT-245

# Conflicts:
#	feature/trainee/home/src/main/java/co/kr/tnt/trainee/home/TraineeHomeScreen.kt

SeonJeongk committed:
[TNT-245] chore: conflict 해결
